cas-server-integration-ignite/src/main/java/org/jasig/cas/ticket/registry/IgniteTicketRegistry.java

```diff
@@ -0,0 +1,317 @@
+        return false;
+    }
+
+    /**
+     * Flag to indicate whether this registry instance should participate in reporting its state with
+     * default value set to {@code true}.
+     *
+     * <p>Therefore, the flag provides a level of flexibility such that depending on the cache and environment
+     * settings, reporting statistics
+     * can be set to false and disabled.</p>
+     *
+     * @param supportRegistryState true, if the registry is to support registry state
+     * @see #sessionCount()
+     * @see #serviceTicketCount()
+     * @see org.jasig.cas.monitor.Monitor
+     */
+    public void setSupportRegistryState(final boolean supportRegistryState) {
+        this.supportRegistryState = supportRegistryState;
+    }
+
+    /**
+     * Init.
+     */
+    @PostConstruct
+    public void init() {
+        logger.info("Setting up Ignite Ticket Registry...");
+
+        if (logger.isDebugEnabled()) {
+            logger.debug("igniteConfiguration.cacheConfiguration={}", igniteConfiguration.getCacheConfiguration());
+            logger.debug("igniteConfiguration.getDiscoverySpi={}", igniteConfiguration.getDiscoverySpi());
+            logger.debug("igniteConfiguration.getSslContextFactory={}", igniteConfiguration.getSslContextFactory());
+            logger.debug("Ticket-granting ticket timeout: [{}s]", this.ticketGrantingTicketTimeoutInSeconds);
+            logger.debug("Service ticket timeout: [{}s]", this.serviceTicketTimeoutInSeconds);
+        }
+
+        if (Ignition.state() == IgniteState.STOPPED) {
+            ignite = Ignition.start(igniteConfiguration);
+        } else if (Ignition.state() == IgniteState.STARTED) {
+            ignite = Ignition.ignite();
+        }
+
+        serviceTicketsCache = ignite.getOrCreateCache(servicesCacheName);
+        serviceTicketsCache.getConfiguration(CacheConfiguration.class)
+            .setExpiryPolicyFactory(CreatedExpiryPolicy.factoryOf(new Duration(TimeUnit.SECONDS, serviceTicketTimeoutInSeconds)));
+
+        ticketGrantingTicketsCache = ignite.getOrCreateCache(ticketsCacheName);
+        ticketGrantingTicketsCache.getConfiguration(CacheConfiguration.class)
+            .setExpiryPolicyFactory(CreatedExpiryPolicy.factoryOf(new Duration(TimeUnit.SECONDS, ticketGrantingTicketTimeoutInSeconds)));
+    }
+
+    @Override
+    public int sessionCount() {
+        return BooleanUtils.toInteger(this.supportRegistryState, this.ticketGrantingTicketsCache
+            .size(CachePeekMode.ALL), super.sessionCount());
+    }
+
+    @Override
+    public int serviceTicketCount() {
+        return BooleanUtils.toInteger(this.supportRegistryState, this.serviceTicketsCache
+            .size(CachePeekMode.ALL), super.serviceTicketCount());
+    }
+
+    /**
+     * Make sure we shutdown Ignite when the context is destroyed.
+     */
+    public void shutdown() {
+        Ignition.stopAll(true);
+    }
+
+    @Override
+    public void destroy() throws Exception {
```

Comment:
That's from starting this work from the EhCache registry code. The other ticket registries should be reviewed for this as well.